Src/Newtonsoft.Json/Linq/JConstructor.cs

```diff
@@ -164,9 +164,10 @@ public override void WriteTo(JsonWriter writer, params JsonConverter[] converter
         {
             writer.WriteStartConstructor(_name);
 
-            foreach (JToken token in Children())
+            int count = _values.Count;
```

Comment:
I did a quick benchmark and the difference is very minor. I'd prefer to use `_values.Count` in the for loop and save this sort of optimization for hot path code.